assets/js/components/KeyMetrics/MetricsSelectionPanel/Footer.js

```diff
@@ -244,28 +246,46 @@ export default function Footer( {
 		setPrevIsOpen( isOpen );
 	}, [ isOpen, prevIsOpen ] );
 
+	let metricsLimitError;
+	if ( selectedMetrics?.length < MIN_SELECTED_METRICS_COUNT ) {
```

Comment:
Just spotted something else - if `selectedMetrics` is `undefined` this condition will return `false` - resulting in the "0 selected (up to 4)" text being displayed instead of "Select at least 2 metrics".

What we should do is create a new variable e.g. `const selectedMetricsCount = selectedMetrics?.length || 0` and use this in place of `selectedMetrics?.length` throughout.